packages/jwk-did-provider/src/jwk-did-provider.ts

```diff
@@ -0,0 +1,119 @@
+ * @public
+ */
+export class JwkDIDProvider extends AbstractIdentifierProvider {
+  private readonly defaultKms: string
+
+  constructor(options: { defaultKms: string }) {
+    super()
+    this.defaultKms = options.defaultKms
+  }
+
+  /** {@inheritDoc @veramo/veramo-core#IDIDManager.didManagerCreate} */
+  async createIdentifier(args: ICreateIdentifierArgs, context: IRequiredContext): Promise<Omit<IIdentifier, 'provider'>> {
+    const key = await this.importProvidedOrGeneratedKey(
+      {
+        kms: args.kms,
+        options: args.options,
+      },
+      context
+    )
+    const jwk = await jose.exportJWK(Uint8Array.from(Buffer.from(key.publicKeyHex, 'hex')))
+
+    if (args.options?.use) {
+      jwk.use = args.options?.use
+    }
+
+    const identifier: Omit<IIdentifier, 'provider'> = {
+      did: `did:jwk:${base64url(JSON.stringify(jwk))}`,
+      controllerKeyId: key.kid,
+      keys: [key],
+      services: [],
+    }
+
+    debug('Created DID: ', identifier.did)
+    return identifier
+  }
+
+  /** {@inheritDoc @veramo/veramo-core#IDIDManager.didManagerDelete} */
+  async deleteIdentifier(identifier: IIdentifier, context: IRequiredContext): Promise<boolean> {
+    // JWKs are entirely in memory constructs, without on chain support. Veramo can store them in the did store, but that is not the responsibility of the DID provider itself.
+    return true
+  }
+
+  /** {@inheritDoc @veramo/veramo-core#IDIDManager.didManagerAddKey} */
+  async addKey(args: IAddKeyArgs, context: IRequiredContext): Promise<any> {
+    return Promise.reject(Error('Not supported for DID JWKs'))
+  }
+
+  /** {@inheritDoc @veramo/veramo-core#IDIDManager.didManagerRemoveKey} */
+  async removeKey(args: IRemoveKeyArgs, context: IRequiredContext): Promise<any> {
+    return Promise.reject(Error('Not supported for DID JWKs'))
+  }
+
+  /** {@inheritDoc @veramo/veramo-core#IDIDManager.didManagerAddService} */
+  async addService(args: IAddServiceArgs, context: IRequiredContext): Promise<any> {
+    return Promise.reject(Error('Not supported for DID JWKs'))
+  }
+
+  /** {@inheritDoc @veramo/veramo-core#IDIDManager.didManagerRemoveService} */
+  async removeService(args: IRemoveKeyArgs, context: IRequiredContext): Promise<any> {
+    return Promise.reject(Error('Not supported for DID JWKs'))
+  }
+
+  /**
+   * We optionally generate and then import our own keys.
+   *
+   * Reason for this is that we want to be able to assign Key IDs (kid), which Veramo supports on import, but not creation. The net result is that we do not support keys which have been created from keyManagerCreate
+   *
+   * @param kms The KMS to use
+   * @param options Allows to set a kid for the new key
+   * @param context The Veramo agent context
+   * @private
+   */
+  private async importProvidedOrGeneratedKey(args: IImportProvidedOrGeneratedKeyArgs, context: IRequiredContext): Promise<IKey> {
+    const kid = args.options?.kid ? args.options.kid : args.options?.key?.kid
```

Comment:
From the DID:JWK spec:

Since did:jwk only contains a single key, the DID URL fragment identifier is always a fixed #0 value.

If the JWK contains a kid value it is not used as the reference, #0 is the only valid value.

So we cannot have a kid argument, It always needs to be #0.

Reply:
removed and set to #0